src/trie/TrieNode.ts

```diff
@@ -0,0 +1,123 @@
+/**
+ * Node of a trie data structure.
+ * The node can be root or any other node in Trie data structure.
+ * It has access to its parent and all of its children.
+ * If the node is the end of a word, it has the flag `isEndOfWord` set to true and `word` is set to the indexed word.
+ * If the node is the end of a word, it can contain some additional data.
+ */
+export default class TrieNode {
```

Comment:
If you make Trie a class with generic type you need to also change it here to `TrieNode<T>` because here data is actually stored.